core/base-service/got.js

```diff
@@ -0,0 +1,96 @@
+'use strict'
+
+const got = require('got')
+const { Inaccessible, InvalidResponse } = require('./errors')
+
+function requestOptions2GotOptions(options) {
+  const requestOptions = Object.assign({}, options)
+  const gotOptions = {}
+  const interchangableOptions = ['body', 'form', 'headers', 'method', 'url']
+
+  interchangableOptions.forEach(function (opt) {
+    if (opt in requestOptions) {
+      gotOptions[opt] = requestOptions[opt]
+      delete requestOptions[opt]
+    }
+  })
+
+  if ('qs' in requestOptions) {
+    gotOptions.searchParams = requestOptions.qs
+    delete requestOptions.qs
+  }
+
+  if ('gzip' in requestOptions) {
+    gotOptions.decompress = requestOptions.gzip
+    delete requestOptions.gzip
+  }
+
+  if ('strictSSL' in requestOptions) {
+    gotOptions.https = {
+      rejectUnauthorized: requestOptions.strictSSL,
+    }
+    delete requestOptions.strictSSL
+  }
+
+  if ('auth' in requestOptions) {
+    gotOptions.username = requestOptions.auth.user
+    gotOptions.password = requestOptions.auth.pass
+    delete requestOptions.auth
+  }
+
+  if (Object.keys(requestOptions).length > 0) {
+    throw new Error(`Found unrecognised options ${Object.keys(requestOptions)}`)
```

Comment:
I've decided to fail loudly if we don't know what to do with it. Obviously this massively constrains what selection of options we can use without modifying this, but I'm pretty certain I've covered the full subset of options we are actively using. If we try to use another one, we need to work out the translation not make an assumption.

Reply:
For some reason I started with this file from `sendRequest` below, and then as I was reading through `requestOptions2GotOptions` found myself wondering why the keys were being deleted on `requestOptions` since it seemed like that was tossed/unused for the actual request but now see why as we're using it for validation.

I don't expect any issues with this, but since it will add processing steps on every request do we want to do some kind of perf run?

Reply:
actually have thinking on this some more, such a hypothetical test would really need to be executed under load over an extended duration which doesn't really seem feasible/viable. i think this will be a case better handled by simply keeping an eye on prod for a while after deployed to be prepared to scale out or even roll back should something really go wildly unexpected

Reply:
I don't think performance should be a worry here. The options object is always going to be a small object with only a few keys.